src/shared/components/case-editor/case-event-completion/case-event-completion.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { StateMachine } from '@edium/fsm';
+import { EventCompletionStateMachineService, WorkAllocationService } from '../services';
+import { SessionStorageService } from '../../../services';
+import { EventCompletionComponentEmitter, EventCompletionStateMachineContext } from '../domain';
+import { EventCompletionParams } from '../domain/event-completion-params.model';
+
+@Component({
+  selector: 'ccd-case-event-completion',
+  templateUrl: './case-event-completion.html'
+})
+export class CaseEventCompletionComponent implements OnChanges, EventCompletionComponentEmitter {
+
+  @Input()
+  public eventCompletionParams: EventCompletionParams;
+
+  @Output()
+  eventCanBeCompleted: EventEmitter<boolean> = new EventEmitter();
+
+  public stateMachine: StateMachine;
+  public context: EventCompletionStateMachineContext;
+
+  constructor(private service: EventCompletionStateMachineService,
+    private readonly router: Router,
+    private readonly route: ActivatedRoute,
+    private readonly sessionStorageService: SessionStorageService,
+    private readonly workAllocationService: WorkAllocationService) {
+  }
+
+  public ngOnChanges(changes?: SimpleChanges): void {
+    if (changes.eventCompletionParams && changes.eventCompletionParams.currentValue) {
```

Comment:
I'm a bit confused here maybe you can clarify. Each time with `eventCompletionParams ` changes, we re-initialise all this code?

Reply:
The eventCompletionParams is an input property and its value is populated in the parent component only when the user clicks on submit button.